In [1]:
# Importing Libraries
import pandas as pd
import numpy as np
import re

#Importing library for stemming
from nltk.stem import PorterStemmer
import string

#Importing library for lemmatizer
from nltk.stem import WordNetLemmatizer

import nltk
from nltk.tokenize import word_tokenize

import warnings
warnings.filterwarnings('ignore')

In [2]:
# To show all rows
pd.set_option('display.max_rows', None)

# To show all columns
pd.set_option('display.max_columns', None)

In [3]:
nltk.download('punkt')
nltk.download('punkt_tab')

[nltk_data] Downloading package punkt to /Users/mukulgarg/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package punkt_tab to
[nltk_data]     /Users/mukulgarg/nltk_data...
[nltk_data]   Package punkt_tab is already up-to-date!


True

In [4]:
#Load Dataset File
#Replace the dataset file path with your own dataset path
dataset = pd.read_csv("data.csv")

In [5]:
#printing first 5 rows of data
dataset.head(5)

Data
0  Watch or listen live weekdays at 8:30am MT at ...
1  Watch or listen live weekdays at 8:30am MT at ...
2               Chubby And Hot, Always Stir The Pot!
3               Chubby And Hot, Always Stir The Pot!
4  Journalist, publisher of Rebel News — telling ...

In [6]:
#Joining all the text data into a single string
text_data = ' '.join(dataset['Data'].astype(str)).lower()
#text_data

In [7]:
#All text data into a single string
#text_data

In [8]:
review = re.sub('[^a-zA-Z0-9]',' ',text_data) #Removing non alpha-numeric characters from the text data
#review

In [9]:
tokenized_words = word_tokenize(review)

In [10]:
#print(tokenized_words)

In [11]:
len(tokenized_words)

122774

## Count of unique words/tokens before stemming/Lemmatization

In [13]:
unique_words = set(tokenized_words)

In [14]:
unique_word_counts = len(unique_words)

In [15]:
#print(unique_words)

In [16]:
unique_word_counts

7893

# Stemming

## Count of unique words/tokens After stemming

In [19]:
# Initializing Porter Stemmer
stemmer = PorterStemmer()

# Applying stemming to each word/token
stemmed_words = [stemmer.stem(token) for token in tokenized_words]

# Count unique tokens after stemming
stemmed_unique_words = set(stemmed_words)
stemmed_unique_words_count = len(stemmed_unique_words)

In [20]:
stemmed_unique_words_count

6528

In [21]:
#print(stemmed_unique_words)

In [22]:
print("Count of Unique words before stemming:",unique_word_counts)
print("Count of Unique words after stemming:",stemmed_unique_words_count)

Count of Unique words before stemming: 7893
Count of Unique words after stemming: 6528


## Q-B1: Use porter stemmer to run stemming. Count the number of unique words/tokens before and after stemming

## Stemming Result:
### As can be seen from above the number of unique words before and after stemming have been reduced from 7893 to 6528.

# Lemmatization

## Count of unique words/tokens after Lemmatization

In [26]:
# Initialize the lemmatizer
lemmatizer = WordNetLemmatizer()

# Applying lemmatization to each word/token
lemma_words = [lemmatizer.lemmatize(token) for token in tokenized_words]

# Count unique tokens after lemmatization
lemma_unique_words = set(lemma_words)
lemma_unique_words_count = len(lemma_unique_words)

In [27]:
#print(lemma_unique_words)

In [28]:
lemma_unique_words_count

7334

In [29]:
print("Count of Unique words before lemmatization:",unique_word_counts)
print("Count of Unique words after lemmatization:",lemma_unique_words_count)

Count of Unique words before lemmatization: 7893
Count of Unique words after lemmatization: 7334


## Q-B2: Perform lemmatization using NLTK lemmatizer . Count the number of unique words/tokens before and after lemmatization?

## Lemmatization Result:
### As can be seen from above the number of unique words before and after lemmatization have been reduced from 7893 to 7334.